### PR TITLE
fix(experiments): Use serializer for feature flag updates as well

### LINKED ISSUE #11070: Experiments with behavioral cohorts can be created
## Bug description

We disabled behavioral cohorts for feature flags, but looks like users can subvert this by using an experiment instead. Not great.

*Please describe.*  
*If this affects the front-end, screenshots would be of great help.*  

## How to reproduce

1. Create an experiment
2. Add behavioral cohort filter. Save.
3. See feature flag with behavioral cohort created

## Environment

- [ ] PostHog Cloud
- [ ] self-hosted PostHog, version/commit: please provide

### PR DESCRIPTION
**Pull request by neilkakkar**
## Problem

Fixes https://github.com/PostHog/posthog/issues/11070

also adds a test for timezone fixes, which should be handled in https://github.com/PostHog/posthog/pull/11935 (I'll keep a track of this, wanted the test in master so I don't have to keep it in my head)

## Changes

👉 *Stay up-to-date with [PostHog coding conventions](https://posthog.com/docs/contribute/coding-conventions) for a smoother review.*

### CODE REVIEW

File: ee/clickhouse/views/test/test_clickhouse_experiments.py [hunk 1]
```diff
@@ -782,6 +837,116 @@ def test_experiment_flow_with_event_results(self):
         self.assertEqual(response_data["significance_code"], ExperimentSignificanceCode.NOT_ENOUGH_EXPOSURE)
         self.assertAlmostEqual(response_data["expected_loss"], 1, places=2)
 
+    @pytest.mark.skip(
+        reason="This test needs proper timezone support, coming in https://github.com/PostHog/posthog/pull/11935"
+    )
```
Comment (EDsCODE) [inline ee/clickhouse/views/test/test_clickhouse_experiments.py:842]: This can be addressed now?

Comment (neilkakkar) [reply ee/clickhouse/views/test/test_clickhouse_experiments.py:842]: Oh, lovely, missed that it had been merged! 👀

... Is that why you were waiting to review 😂

**Review by EDsCODE — APPROVED** [review]
lgtm besides the skipped test

**Comment by neilkakkar** [comment]
Will address the test in a follow up PR, seems cleaner (not a small change anymore)

**Pull request merged**